Common/Data/Market/TradeBar.cs

```diff
@@ -613,6 +616,47 @@ public static TradeBar ParseForex(SubscriptionDataConfig config, StreamReader st
             ParseForex(tradeBar, config, streamReader, date);
             return tradeBar;
         }
+        
+        /// <summary>
+        /// Parses Index trade bar data into the specified tradebar type, useful for custom types with OHLCV data deriving from TradeBar
+        /// </summary>
+        /// <typeparam name="T">The requested output type, must derive from TradeBar</typeparam>
+        /// <param name="config">Symbols, Resolution, DataType, </param>
+        /// <param name="line">Line from the data file requested</param>
+        /// <param name="date">The base data used to compute the time of the bar since the line specifies a milliseconds since midnight</param>
+        /// <returns></returns>
+        public static T ParseIndex<T>(SubscriptionDataConfig config, string line, DateTime date)
+            where T : TradeBar, new()
+        {
+            // CFD has the same data format as Forex
+            return ParseForex<T>(config, line, date);
+        }
+
+        /// <summary>
+        /// Parses Index trade bar data into the specified tradebar type, useful for custom types with OHLCV data deriving from TradeBar
+        /// </summary>
+        /// <param name="config">Symbols, Resolution, DataType, </param>
+        /// <param name="line">Line from the data file requested</param>
+        /// <param name="date">The base data used to compute the time of the bar since the line specifies a milliseconds since midnight</param>
+        /// <returns></returns>
+        public static TradeBar ParseIndex(SubscriptionDataConfig config, string line, DateTime date)
+        {
+            // CFD has the same data format as Forex
+            return ParseForex(config, line, date);
```

Comment:
The index data is time, OHLC, and the "volume"/ticks are not really useful (fixed 60 per minute). Probably best to make its own parser.